### ISSUE TITLE
`get_predicted.glmmTMB(predict="expectation")` in zero-inflated models

### ISSUE THREAD

**Issue by vincentarelbundock** [easystats/insight#451]
`get_predicted.glmmTMB(model, predict="expectation")` does not produce the same results as `predict.glmmTMB(model, type="response")` for zero inflated models.

The documentation in the `predict` method states:

> Denoting mu as the mean of the conditional distribution and ‘p’ as the zero-inflation probability.
> “response” expected value; this is mu\*(1-p) for zero-inflated models and ‘mu’ otherwise

Our own method only takes `mu`:

``` r
library(glmmTMB)
library(insight)

data("fish")
m1 <- glmmTMB(
  count ~ child + camper + (1 | persons),
  ziformula = ~ child + camper + (1 | persons),
  data = fish,
  family = truncated_poisson())

get_predicted(m1, predict = "expectation") |> head()
# [1] 1.023045 2.130428 1.023045 1.663721 1.023045 2.633594

predict(m1, type = "response") |> head()
# [1] 0.2400130 0.9726394 0.2400130 0.4613930 0.2400130 0.6362097
```

We can get the same results by applying the formula in the `glmmTMB` docs

``` r
pred <- get_predicted(m1, predict = "expectation") * (1 - predict(m1, type = "zprob"))

head(pred)
# [1] 0.2400130 0.9726394 0.2400130 0.4613930 0.2400130 0.6362097
```

Should we check if the model is zero inflated and do this correction automatically?

**Comment by bwiernik** [comment]
Yes, expectation should incorporate the zero inflation part. 

As you are working through these, can you make a list of options aside from link/expectation/prediction/classification that various models support? (Eg, conditional response)

**Comment by bwiernik** [comment]
I can investigate in the next few weeks

**Comment by bwiernik** [comment]
A few first thoughts:

1. Zero inflated and hurdle models don't have clean likelihoods. They Franken-models with 2 different likelihoods stapled together. That means they don't have a clean link function. 
2. I wonder what the zeroinfl package does for its intervals? (If it has them)
3. Given (1), I think the 2 main options would be (a) assume normality and use the delta method SEs returned from glmmTMB; or (b) use a parametric bootstrap for it. A third option (c) might be to adapt methods from merTools, but not sure if that's feasible given major differences in the two estimators. 
4. This is one of the main reasons to use MCMC for stuff...

**Comment by strengejacke** [comment]
See section _Details_ here: https://strengejacke.github.io/ggeffects/reference/ggpredict.html
In particular sub-section "Zero-Inflated and Zero-Inflated Mixed Models with glmmTMB".

And see also here: https://strengejacke.github.io/ggeffects/articles/introduction_randomeffects.html#marginal-effects-for-zero-inflated-mixed-models-1

Reference: Brooks ME, Kristensen K, Benthem KJ van, Magnusson A, Berg CW, Nielsen A, et al. glmmTMB Balances Speed and Flexibility Among Packages for Zero-inflated Generalized Linear Mixed Modeling. The R Journal. 2017;9: 378–400.

My implementation in ggeffects:
https://github.com/strengejacke/ggeffects/blob/master/R/get_predictions_glmmTMB.R
https://github.com/strengejacke/ggeffects/blob/master/R/predict_zero_inflation.R

**Comment by vincentarelbundock** [comment]
FWIW, I think we should report the standard errors returned by the package's `predict` method instead of making additional or different statistical assumptions to compute our own idiosyncratic SEs.

**Comment by bwiernik** [comment]
I don't think we should report the standard errors. Those will be misinterpreted as confidence intervals.

**Comment by strengejacke** [comment]
should be resolved in #501 to #503
